### PR TITLE
[TASK-884] Implement reusable export button

### PR DESCRIPTION
**Pull request by pauloamorimbr**
[Task](https://www.notion.so/kobotoolbox/Add-reusable-export-button-ce3a0d358d95437f883ee64ea1a3cb20?pvs=4)

## Checklist

1. [x] If you've added code that should be tested, add tests
2. [x] If you've changed APIs, update (or create!) the documentation
3. [x] Ensure the tests pass
4. [x] Run `./python-format.sh` to make sure that your code lints and that you've followed [our coding style](https://github.com/kobotoolbox/kpi/blob/main/CONTRIBUTING.md)
5. [x] Write a title and, if necessary, a description of your work suitable for publishing in our [release notes](https://community.kobotoolbox.org/tag/release-notes)
6. [x] Mention any related issues in this repository (as #ISSUE) and in other repositories (as kobotoolbox/other#ISSUE)
7. [ ] Open an issue in the [docs](https://github.com/kobotoolbox/docs/issues/new) if there are UI/UX changes
8. [x] Create a testing plan for the reviewer and add it to the Testing section

## Description

This PR implements the functionality for the export button.
- Actual export function is mocked and will not generate any data
- The mocked results now have a 50% chance of failure for testing purposes

## Notes

- A new "ExportButton" component was created.
- It receives an `async` function to be called when clicked
- If the export process is successfully started it will display a message to the user
- If an error occurs, `alertify` is used to display an error message

## Testing

- Access the project settings
- Be sure to have the `?ff_activityLogsEnabled=true` feature flag on
- Access the `Activity` sub view
- Click the export button some times.
- It should display the successful or error message depending on the (random) result.

### CODE REVIEW

File: jsapp/js/components/exportButton/exportButton.component.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    title={t('Exporting data')}
+    buttons={[
+      {
+        label: 'Ok',
+        onClick: onClose,
+      },
+    ]}
+  >
+    <div>
+      <p>
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/exportButton/exportButton.component.tsx:20]: And if you take a look at `KoboPrompt`, then you'll see that we are wrapping `props.children` inside a wrapper :)

File: jsapp/js/components/exportButton/exportButton.component.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          "Your export request is currently being processed. Once the export is complete, you'll receive an email with all the details."
+        )}
+      </p>
+    </div>
+  </KoboPrompt>
+);
+
+export default function ExportToEmailButton({
+  exportFunction,
+  ...props
```
[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/exportButton/exportButton.component.tsx:31]: If we end up only exposing `label`, than the stories file needs to only expose it too :)

File: jsapp/js/components/exportButton/exportButton.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import type {Meta, StoryFn} from '@storybook/react';
+
+import ExportButton from './exportButton.component';
+
+import buttonStory from '../common/button.stories';
+
+export default {
+  title: 'common/ExportButton',
```
Comment (magicznyleszek) [inline jsapp/js/components/exportButton/exportButton.stories.tsx:8]: Please change it to `misc/ExportButton`, to not put it in the "common" section.

File: jsapp/js/components/exportButton/exportButton.component.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (pauloamorimbr) [reply jsapp/js/components/exportButton/exportButton.component.tsx:31]: At first I exposed only label, but then I thought someone could want to chnage other properties from the button, like the variant or size, according to the needed layout, so I changed the implementation to allow for more control.
Do you think it's better to change expose label for now and, if needed, someone can extend it in the future?

[the remark this answers is not in the document]
Comment (magicznyleszek) [reply jsapp/js/components/exportButton/exportButton.component.tsx]: Yes, let's expose only `label`. I don't see us (based on Figma designs for all the features that we know will use this) wanting anything else in the future

**Review by magicznyleszek — APPROVED** [review]

**Pull request merged**